lib/cli.js

```diff
@@ -397,6 +408,10 @@ module.exports = async (argv) => {
 			: path.resolve(process.cwd(), cli.flags.configBasedir);
 	}
 
+	if (cli.flags.globbyOptions) {
+		optionsBase.globbyOptions = cli.flags.globbyOptions;
```

Comment:
[suggest] What do you think about the following idea to pass JSON value?

```suggestion
		const value = cli.flags.globbyOptions;

		try {
			optionsBase.globbyOptions = JSON.parse(value);
		} catch (_) {
			process.stderr.write(
				`Invalid option ${red('"--globby-options"')}. The value ${red(
					`"${value}"`,
				)} is not valid JSON.${EOL}`,
			);
			process.exit(EXIT_CODE_ERROR); // eslint-disable-line no-process-exit
		}
```

For example:

<img width="862" alt="image" src="https://user-images.githubusercontent.com/473530/199236640-794f6869-0671-40e1-95c9-22312d45b79f.png">

Reply:
yes, JSON sounds promising!

just a small doubt do we always need to **stringify** and **parse** the `value` or will there be any case where this can fail!!?? 🤔

Reply:
> just a small doubt do we always need to stringify or parse the `value` or will there be any case where this can fail!!?? 🤔

I think `JSON.parse` is a way to convert a string to an object. Do you have another way?

> Also this change is causing an error in `cli.test.js file`. not sure if i have update anything their based on this change!

What kind of errors?

Reply:
<img width="826" alt="Screenshot 2022-11-02 at 7 50 24 AM" src="https://user-images.githubusercontent.com/10995431/199380290-34fa6e77-22a8-4728-b465-7b88fae4bc17.png">

When running `jest` command getting this error! @ybiquitous

Reply:
@sidverma32 You may need to update failed [snapshots](https://jestjs.io/docs/snapshot-testing) by the `npx jest -u` command. Can you please try it?

Reply:
oh yeah! i did last time to fix the same ..my bad :)

Updated!

Reply:
Thanks. Additionally, can you please add the following test cases to `cli.test.js`?

- Success case: `--globby-options` should work on actual file system, e.g. `--globby-options '{"dot": true}'` should scan files in dot directories
- Failure case: `--globby-options` with an invalid JSON value should fail with the expected message and exit status

Reply:
> Thanks. Additionally, can you please add the following test cases to `cli.test.js`?
> 
> * Success case: `--globby-options` should work on actual file system, e.g. `--globby-options '{"dot": true}'` should scan files in dot directories
> * Failure case: `--globby-options` with an invalid JSON value should fail with the expected message and exit status

```
it('flags.globbyOptions', () => {
		//success
		expect(buildCLI(['--globby-options={"dot": true}']).flags.globbyOptions).toBe('/path/to/dir');
		expect(buildCLI(['--go={"dot": true}']).flags.globbyOptions).toBe('/path/to/dir');
		
		//failure
		expect(buildCLI(['--globby-options={"dot": foo}']).flags.globbyOptions).toBe(
			expect(process.exit).toHaveBeenCalledWith(0);
			expect(console.log).toHaveBeenCalledWith(
			   expect.stringContaining('Invalid JSON value'),
		    );
		 );
	  );
	});
```

Is this fine?

Reply:
No, please keep the `flags.globbyOptions` case. Let's add new cases to test `process.exitCode`, `process.stdout`, or `process.stderr` like this:

https://github.com/stylelint/stylelint/blob/042d1adee8a15f61d4d3fdf5c2e0e38569303942/lib/__tests__/cli.test.js#L364-L376

Reply:
```
`it('output a message when wrong --globby-options provided', async () => {
		await cli([
			`--globby-options=stylelint-test-globby-options`,
			'--config',
			fixturesPath('default-severity-warning.json'),
			fixturesPath('empty-block.css'),
		]);

		expect(process.exitCode).toBe(2);

		expect(process.stdout.write).toHaveBeenCalledTimes(1);

		//expect((process.stdout.write).toBe('Invalid JSON value provided.');
	});

`
```
as per my understanding from your suggestion, added this code-block . Please guide me how to print "Invalid" message as it was giving me error. (I am beginner in `jest`  )

Reply:
The expectation code should be as follows:

```js
expect(process.exitCode).toBe(2);
expect(process.stdout.write).toHaveBeenCalledTimes(0);
expect(process.stderr.write).toHaveBeenCalledTimes(1);
expect(stripAnsi(process.stderr.write.mock.calls[0][0])).toContain(...);
```